controllers/rollingupgrade_controller.go

```diff
@@ -464,11 +461,15 @@ func (r *RollingUpgradeReconciler) populateAsg(ruObj *upgrademgrv1alpha1.Rolling
 		return errors.New("No ASG found")
 	} else if len(result.AutoScalingGroups) > 1 {
 		log.Printf("%s: Too many asgs found with name %d!\n", ruObj.Name, len(result.AutoScalingGroups))
-		return errors.New("Too many asgs")
+		return errors.New("Too many ASGs")
 	}
 
 	asg := result.AutoScalingGroups[0]
-	r.Asg = asg
+	if r.Asgs == nil {
+		// Init map to hold all parallel reconciling ASGs
+		r.Asgs = make(map[string]*autoscaling.Group)
+	}
+	r.Asgs[ruObj.Name] = asg
 	r.ruObjNameToASG.Store(ruObj.Name, asg)
```

Comment:
I see it already gets saved here.
Lets just get rid of `r.Asgs` it is not needed at all when we store it everything in a sync.Map `ruObjNameToASG`

Reply:
yes, good point! I'll change all uses of `r.Asg` (`r.Asgs` after my changes) to leverage this store you're highlighting